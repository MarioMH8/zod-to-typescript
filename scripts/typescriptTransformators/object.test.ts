import { ZodToTypescript } from "@scripts/index";
import { z as zod } from "zod";

describe("object", () => {
	it("object", () => {
		const result = ZodToTypescript.convert(
			zod.object({
				name: zod.string(),
				age: zod.number(),
			}),
		);
		expect(result).toMatchSnapshot();
	});

	it("object with key optional", () => {
		const result = ZodToTypescript.convert(
			zod.object({
				name: zod.string(),
				age: zod.undefined(),
			}),
		);
		expect(result).toMatchSnapshot();
	});

	it("object with union undefined", () => {
		const result = ZodToTypescript.convert(
			zod.object({
				name: zod.string(),
				age: zod.number(),
				optional: zod.union([zod.undefined(), zod.string()]),
			}),
		);

		expect(result).toMatchSnapshot();
	});

	it("good template key", () => {
		const result = ZodToTypescript.convert(
			zod.object({
				"test-1": zod.string(),
				test: zod.string(),
				"test@": zod.string(),
				111: zod.string(),
			}),
		);

		expect(result).toMatchSnapshot();
	});
});
