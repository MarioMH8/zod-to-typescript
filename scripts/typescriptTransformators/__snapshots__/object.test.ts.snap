// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`object > object 1`] = `
"type Zod2ts_0_duplojs = {
    name: string;
    age: number;
};"
`;

exports[`object > object with key optional 1`] = `
"type Zod2ts_1_duplojs = {
    name: string;
    age?: undefined;
};"
`;

exports[`object > object with union undefined 1`] = `
"type Zod2ts_2_duplojs = {
    name: string;
    age: number;
    optional?: undefined | string;
};"
`;
